### ISSUE TITLE
Waiting to connect on one tab/window should not lock up the entire application.

### ISSUE THREAD

**Issue by amfriedman** [Sequel-Ace/Sequel-Ace#1945]
It is common to have multiple database connections open across multiple tabs or windows. Sometimes one of these tabs or windows will have a connectivity issue (e.g. I moved my location and so my new IP is no longer whitelisted so I am blocked by that DB's firewall). When this occurs, the current behavior is that the entire Sequel Ace app locks up or freezes while this one connection waits to re-establish. This is effectively a blocking bug for the user; I can't choose to navigate to another tab or window to work on another task with another DB. If the timeout is long, then the app is effectively frozen for 30-60 seconds or more, which is highly frustrating.

It would be a far superior UX if these connection hangs were siloed or isolated to their own tab/window, and only make themselves known when the user has that tab or window in focus. This would free the user up to work on another tab/window and stay productive, and happy. 

Thanks!
